### PR TITLE
Feature/build json

### PR DESCRIPTION
**Pull request by ciaradunks**
## Summary of the discussion

Incorporates the ability to automatically generate datapackage.json files from input CSV files. 

- In the case of only using existing facades (from oemof-tabular), the building.infer_metadata function can be used directly, with the foreign keys manually inputted as arguments in scripts/infer.py
- In the case of using new facades (from oemof-tabular-plugins), an additional function is implemented to add foreign keys to the datapackage.json file

### Added
  - New function for adding additional foreign keys to datapackage.json in src/oemof_tabular_plugins/datapackage/building.py
  - Added the use of the function in examples/scenarios/wefe_pv_panel/scripts/infer.py

### Changed
- The scripts/infer.py files for all complete examples and ensured they run through correctly

### CODE REVIEW

File: examples/scenarios/wefe_pv_panel/scripts/infer.py [hunk 1]
```diff
@@ -26,4 +28,15 @@
     **kwargs,
 )
 
-# run building.infer, reopen json file and ammend datapackage
\ No newline at end of file
+# this function adds any additional foreign keys to the datapackage.json file
+otp_building.add_foreign_keys_to_datapackage(
+    "datapackage.json",
+    {
```
Comment (Bachibouzouk) [inline examples/scenarios/wefe_pv_panel/scripts/infer.py:34]: This is a good first step, but I think we can infer the `from_bus` and `to_bus` bits automatically inside the `[rl-institut/oemof-tabular-plugins]` function I would say

File: examples/scenarios/wefe_pv_panel/scripts/infer.py [hunk 2]
```diff
@@ -26,4 +28,15 @@
...
+        "pv_panel": {
+            "from_bus": {"resource": "bus", "fields": "name"},
+            "to_bus": {"resource": "bus", "fields": "name"},
+            "t_air": {"resource": "pv_panel_profile"},
```
Comment (Bachibouzouk) [inline examples/scenarios/wefe_pv_panel/scripts/infer.py:38]: Looking at all values of the `name` field of all `schema.fields` field for all resources which have `data/sequences` as value of their `path` field let us find the information necessary to infer `"t_air": {"resource": "pv_panel_profile"},` as well

**Review by Bachibouzouk — APPROVED** [review]
Nice job, I made a comment on how to make the function fully automatic after the next PR were you lint and change the structure

File: examples/scenarios/wefe_pv_panel/scripts/infer.py (hunk 1 above)
Comment (ciaradunks) [reply examples/scenarios/wefe_pv_panel/scripts/infer.py:34]: yes that's what I tried to do here: `"from_to_bus": ["pv-panel"]`
but for some reason it doesn't add to the pv-panel component

File: examples/scenarios/wefe_pv_panel/scripts/infer.py (hunk 2 above)
Comment (ciaradunks) [reply examples/scenarios/wefe_pv_panel/scripts/infer.py:38]: ah yeah that makes sense, so have it e.g. like:
```        "pv_panel": {
            "from_bus": {"resource": "bus", "fields": "name"},
            "to_bus": {"resource": "bus", "fields": "name"},
            "t_air": {"fields": "name"}, ```

**Pull request merged**